synapse/storage/databases/main/pusher.py

```diff
@@ -478,6 +478,53 @@ def delete_pusher_txn(txn, stream_id):
                 "delete_pusher", delete_pusher_txn, stream_id
             )
 
+    async def delete_all_pushers_with_pushkey_and_user_id(
```

Comment:
I realise in the linked issue that we should add another method for deleting pushers, but I've just thought that since there's a unique constraint on the `pushers` table for `(app_id, push_key, user_name)`, we may be able to get away with just using `delete_pusher_by_app_id_pushkey_user_id`, as app_id should always be `m.email` in this context.

Reply:
We should also probably make sure we remove only entries with `app_id` = `m.email` anyhow I suppose. Note that I don't believe msisdns (phone numbers) are specced here at all yet: https://spec.matrix.org/unstable/client-server-api/#post_matrixclientr0pushersset